motiondetect.py

```diff
@@ -0,0 +1,75 @@
+import cv2 as cv
+from cv2 import imshow
+
+#works for images, recordings, live video 
+def rescaleFrame(frame, scale):
+    #scale the width and height, (cast to int)
+    width = int(frame.shape[1] * scale)
+    height =int(frame.shape[0] * scale)
+    dimensions = (width, height)
+    #return resize frame to particular dimension
+    return cv.resize(frame, dimensions, interpolation=cv.INTER_AREA)
+
+#convert frame to grey, compute Gaussian blur for noise reduction, update ave,
+#compute weighted averages, compute difference between wieghted ave and grayscale frame to get delta (background bodel - grayscale frame)
+def imgProcess(frame, avg):
+    #convert to grayscale image
+    gray = cv.cvtColor(frame, cv.COLOR_BGR2GRAY)
+    #blur image to reduce noise (filter size 21X21)
+    gray = cv.GaussianBlur(gray, (11,11),0)
+    #if first loop, need to set avg
+    if avg is None:
+        avg = gray.copy().astype("float")
+    #get weighted average of prev frame
+    cv.accumulateWeighted(gray, avg, 0.5)
+    #compute difference between first frame and cur frame
+    frameDelta = cv.absdiff(gray, cv.convertScaleAbs(avg))
+    thresh = cv.threshold(frameDelta, 5, 255, cv.THRESH_BINARY)[1]
+    #dilate the threshold image to fill in holes
+    thresh = cv.dilate(thresh, None, iterations=2)
+    #get contours
+    cnts = cv.findContours(thresh.copy(), cv.RETR_EXTERNAL, cv.CHAIN_APPROX_SIMPLE)
+    if len(cnts) == 2:
+        cnts = cnts[0]
+    elif len(cnts) == 3:
+        cnts = cnts[1]
```

Comment:
Please change hardcoded integer/double/float values for variable names for reusablitiy.